WL-3992 Added option to display root folder for the resources.

ouit0408 opened the pull request:
Have added attribute 'displayRootDirectory' which default is set to false
but when set true will display the root folder.
Should not break Lawrence's code.

buckett commented on library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js:
```diff
@@ -28,9 +29,20 @@ $.fn.folderListing = function(options) {
     var html = $('<ul/>').addClass('jqueryFileTree').hide();
-
+    var urlRoot = json[0].resourceId.trim();
+    if(settings.displayRootDirectory && urlRoot === root ){
```
Better to extract this out as a variable, especially as the same test is done in 2 places and they are a long way apart in the code.

buckett replied:
Actually, probably better to have a `container` variable, and by default set it to be equal to `html` but if we are showing the root folder set container equal to `ul`. That way you keep the switching all in the same place.

ouit0408 replied:
when we are showing root folder we want both 'html' and 'ul', as this 'ul' is new list for the root, I am not sure if we can use just one variable

buckett commented on library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js:
```diff
@@ -102,7 +114,12 @@ $.fn.folderListing = function(options) {
 
       li.append(a);
-      html.append(li);
+      if(settings.displayRootDirectory && urlRoot === root){
```
Then here just do `container.append(li)` and it doesn't matter how you got here.

ouit0408 committed:
WL-3992 Added option to display root folder for the resources.

Have added attribute 'displayRootDirectory' which default is set to false
but when set true will display the root folder.
Review Comments Implemented.

ouit0408 commented:
Implemented review comments and have pushed the code .
